src/components/calcite-button/calcite-button.tsx

```diff
@@ -137,18 +137,21 @@ export class CalciteButton {
       </span>
     );
 
+    const className = this.hasContent ? { ["class"]: CSS.contentSlotted } : null;
+
     return (
-      <Host dir={dir} hasText={this.hasText}>
+      <Host dir={dir}>
         <Tag
           {...attributes}
+          {...className}
```

Comment:
Can you just do:

```
<Tag
class={{[CSS. contentSlotted]: this.hasContent}}
```

?

Reply:
We could - it leaves an empty `class` attribute if `hasContent` is false though:
![Screen Shot 2021-01-07 at 5 06 30 PM](https://user-images.githubusercontent.com/4733155/103962059-d0d3db00-510a-11eb-82c4-c8d3ff79e6dd.png)

Reply:
what do you think @jcfranco? I think the empty class is ok but it's nitpicky.

Reply:
I think the stencil way to do this is:

```jsx
<Tag class={{ [CSS.contentSlotted]: this.hasContent }}>
```

ie. conditionally add the `CSS.contentSlotted` class if `this.hasContent` is true.